src/common/snippets/src/lowered/pass/insert_reg_spills.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/insert_reg_spills.hpp"
+
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/op/reg_spill.hpp"
+#include "snippets/op/brgemm.hpp"
+#include "snippets/itt.hpp"
+#include "snippets/utils/utils.hpp"
+
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+
+bool InsertRegSpills::run(LinearIR& linear_ir) {
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::InsertRegSpills")
+
+    auto needs_reg_spill = [](const ExpressionPtr& expr) {
+        return ov::is_type<snippets::op::Brgemm>(expr->get_node());
+    };
```

Comment:
Should it be hidden in target machine?
I think target machine should know if the node (emitter) needs to reg spills.